### ISSUE TITLE
Graphic not scaled correctly (huge)

### ISSUE THREAD

**Issue by markusmaresch** [gnosygnu/xowa#38]
The chart for distribution of votes is dominating the screen - way to big.
Xowa: de.wikipedia.org/wiki/Hausmannstätten
Online: https://de.wikipedia.org/wiki/Hausmannst%C3%A4tten

Version 3.1.3.1. OS-X

**Comment by markusmaresch** [comment]
Same for Vienna and Salzburg: 
de.wikipedia.org/wiki/Wien
de.wikipedia.org/wiki/Salzburg

Similarly, the clima data chart for Paris: de.wikipedia.org/wiki/Paris

**Comment by gnosygnu** [comment]
Hey, thanks for the clear report!

I wasn't able to reproduce it now. I tried using the wiki from [2016-01](https://dumps.wikimedia.or/dewiki/20160111/) as well as the 2015-09 one. See the attached screenshot.

I also checked Wien, Salzburg and Paris. They all look fine.

This may be an OS issue. Can you post a screenshot from your side?

Also, a few other questions:
* What is the date of the dump: 2016-01?
* Which version of XOWA are you using (I'm on v3.1.3.1)
* Are you using files from the XOWA file databases or are you downloading them from online? (shouldn't matter, but just want to make sure)

Thanks!

![dewiki](https://cloud.githubusercontent.com/assets/4994104/12533160/40ff3af0-c1f5-11e5-93b6-4217b961c9a8.PNG)

**Comment by markusmaresch** [comment]
Attached is a screen-shot - on a 4k monitor (really!) - in order to show the issue. Width = 2.6k pixel
System is latest OS-X, Xowa is 3.1.3.1
de.wikipedia.org (from: home/wiki/Help:Wiki_maintenance )
de.wikipedia.org 	n 	2015-08-27 	2015-08-06 	Dump complete 	sqlite3 	v2 	v2 
I believe I used the pre-build 7z and did not import the dump.

![xowa_de_hsmstn](https://cloud.githubusercontent.com/assets/13298571/12537223/08f35bd0-c2ba-11e5-8aa3-f0f421759d94.png)

Which (external) tool is creating the graphic - I can dig and debug more there ..

**Comment by gnosygnu** [comment]
Wow! Monitor looks great. ;)

Three things:

(1)  I tried the HTML on an iMac on my side, and it still looks fine. I attach it below. You can try this on your machine by opening up the html file in a text editor and changing the href on this tag:

```
<base href="file:///D:/xowa/"/>
```

For example, if you run XOWA out of "/Users/markusmaresch/xowa/", it should be

```
<base href="file:///Users/markusmaresch/xowa/"/>
```

(2) I'm downloading the 2015-08-06 dump and seeing if it's an issue with the dump.

(3) Also, when you get a chance can you send me the HTML on your side?

* Click "HTML" near the upper-right part of the screen
* Copy and paste the contents of the HTML and attach it here

Finally, the graphic is actually just multiple divs with the border-radius HTML. No complicated tool involved. XOWA may not be generating the HTML correctly, but I'm surprised that I'm not seeing this on my side. I'll know more when I see your HTML

Anyway, will let you know later today. Thanks!

[dewiki.txt](https://github.com/gnosygnu/xowa/files/102451/dewiki.txt)

**Comment by gnosygnu** [comment]
Just so you know. I pulled down the 2015-08-06 dump, and it also looks fine there.

I'd like to see your HTML from (3). I'm hoping there's something visibly different there.

**Comment by markusmaresch** [comment]
Thanks for the debug instructions - finally I was able to provide more.
1) I used your dewiki.txt - and it displays correctly on my system with all browsers tested
2) I created dewiki_broken_on_mac.txt - it expectedly does not display correctly.
Used tkdiff to compare - there is some difference, but you probably can tell more ..
Thanks again !
[dewiki_broken_on_mac.txt](https://github.com/gnosygnu/xowa/files/111404/dewiki_broken_on_mac.txt)
[dewiki.txt](https://github.com/gnosygnu/xowa/files/111406/dewiki.txt)

**Comment by gnosygnu** [comment]
Thanks! That file was extremely helpful!

It turned out that the widths were 10 times larger than they should have been. I explain more at https://gnosygnu.github.io/xowa/blog.html and I also excerpt below.

This was fixed in tonight's release: https://github.com/gnosygnu/xowa/releases/ . Please give it a try and let me know if it still doesn't work.

Also, I like to credit users for reporting bugs. You get a line in the XOWA Change log with Core: "Parse decimals using US locale, not user locale. Resolves extremely wide pie charts on many de.w pages. {detected by anonymous}". Let me know if you would like to be cited by username, or would rather remain anonymous.

Let me know if there is anything else. Thanks!

## (excerpt)
There were HTML issues in German Wikipedia because all the wikis rely implicitly on American formatting rules for decimals. For example:

* The German Wikipedia page for Hausmannstätten specifies a pie chart with a width (Breite) of 0.9 . See: https://de.wikipedia.org/w/index.php?title=Hausmannst%C3%A4tten&action=edit
* This value gets multiplied by various numbers in the template. For example: <code>width:&#123;&#123;#expr:18*&#123;&#123;&#123;Breite}}} }}em;</code>. See: https://de.wikipedia.org/w/index.php?title=Vorlage:Sitzverteilung/&action=edit

XOWA was doing this multiplication using the user's locale.
* For US users (like me), <code>18\*.9</code> was calculated correctly as <code>16.2</code>
* For German users though, <code>18\*.9</code> was calculated incorrectly as <code>162</code>.
** This was because Java was interpreting the <code>.9</code> using German number-format rules. Since "." is the thousands separator, and "," is the decimal separator, <code>.9</code> became <code>9</code>. For US users, imagine <code>,9</code> becoming <code>9</code>.

v3.1.5 fixes this issue by asserting that all decimal-parsing is done in the US locale.

**Comment by markusmaresch** [comment]
Confirmed fix with 3.1.5.1
